src/activity_handlers.cpp

```diff
@@ -242,7 +256,83 @@ void set_up_butchery( player_activity &act, player &u )
         return;
     }
 
-    const mtype *corpse = items[act.index].get_mtype();
+    bool has_table_nearby = false;
+    for( const tripoint &pt : g->m.points_in_radius( u.pos(), 2 ) ) {
+        if( g->m.has_flag_furn( "FLAT_FURN", pt ) || ( g->m.veh_at( pt ) && g->m.veh_at( pt )->vehicle().has_part( "KITCHEN" ) ) ) {
+            has_table_nearby = true;
+        }
+    }
+    // workshop butchery (full) prequisites
+    if( action == BUTCHER_FULL ) {
+            if( !g->m.has_flag_furn( "BUTCHER_EQ", u.pos() ) ) {
+                u.add_msg_if_player( m_info, _( "You need a butchering rack to perform a full butchery." ) );
+                act.set_to_null();
+                return;
+            }
+            if ( !has_table_nearby ) {
+                u.add_msg_if_player( m_info, _( "You need a table nearby or something else with a flat surface to perform a full butchery." ) );
+                act.set_to_null();
+                return;
+            }
+            if( ( !u.has_quality( quality_id( "CUT" ) ) || !u.has_quality( quality_id( "CUT_FINE" ) ) ) &&
+                ( !u.has_quality( quality_id( "SAW_W" ) ) || !u.has_quality( quality_id( "SAW_M" ) ) ) ) {
+                u.add_msg_if_player( m_info, _( "You need tools that can cut and saw to perform a full butchery." ) );
```

Comment:
A better catch-all statement since almost all fine cutting tools have cutting:
`You needs tool with fine cutting and sawing to perform a full butchery.`

Should help reduce the amount of confusion when players are trying to extract CBMs.